tools/Dashboard/DevHome.Dashboard/Services/WidgetIconService.cs

```diff
@@ -18,103 +17,61 @@ public class WidgetIconService : IWidgetIconService
 {
     private readonly WindowEx _windowEx;
 
-    private readonly IWidgetHostingService _widgetHostingService;
-
     private readonly Dictionary<string, BitmapImage> _widgetLightIconCache;

```

Comment:
Nit: Would it be helpful to decouple and promote the icon caching into a separate service e.g. `WidgetIconCache`?
I am thinking that will make it transparent to the user/caller on how the cache manages storing icon per theme. It will also better allow adding `lock` around the code that manipulates each theme dictionary as well as synching both dictionaries on `remove()`. Another plus would be if we want to add more cache related logging.

Reply:
I second this idea, but for different reasons.  Every instance the cache collection uses both the light cache and the dark cache.  Someone in the future might not remember to perform an operation on the correct cache.  Separating the caching logic into another class would isolate the "If dark, use dark cache, otherwise use light"

Reply:
#2482